File: frontend/src/lib/knowledge.ts
import { api } from "./api";

export type DocStatus = "processing" | "ready" | "failed";

export interface KnowledgeDoc {
  id: string;
  filename: string;
  status: DocStatus;
  error_message: string | null;
  created_at: string;
  chunk_count: number;
}

export const knowledgeApi = {
  list: () => api.get<KnowledgeDoc[]>("/knowledge"),
  upload: (file: File) => api.upload<KnowledgeDoc>("/knowledge/upload", file, file.name),
  remove: (id: string) => api.del<void>(`/knowledge/${id}`),
};

File: backend/app/schemas/knowledge.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.knowledge import DocStatus


class KnowledgeDocOut(BaseModel):
    id: uuid.UUID
    filename: str
    status: DocStatus
    error_message: str | None
    created_at: datetime
    chunk_count: int = 0

    model_config = {"from_attributes": True}
